licensing/license.go

```diff
@@ -25,13 +25,25 @@ import (
 	"os"
 	"reflect"
 	"strings"
+	"sync"
 )
 
 var (
 	startPrefixes = []string{"// Copyright", "// copyright", "// Licensed", "// licensed", "// ELASTICSEARCH CONFIDENTIAL"}
 	endPrefixes   = []string{"package ", "// Package ", "// +build ", "// Code generated", "// code generated", "//go:"}
 
 	errHeaderIsTooShort = errors.New("header is too short")
+
+	// Supported licenses size are: 173, 232, 240, 636 and 745.
+	// We use a buffer of 768 to make sure everything fit
+	// without any additional allocation.
+	defaulBufSize = 768
```

Comment:
Maybe it'd be worth it to iterate over `Headers` and use the length of the biggest as `defaultBufSize` in case we add more licenses and forget to update this number.

Reply:
Good point! Fixed!